benchmarks/expected-plans/q20.txt

```diff
@@ -1,17 +1,17 @@
 Sort: supplier.s_name ASC NULLS LAST
   Projection: supplier.s_name, supplier.s_address
-    LeftSemi Join: supplier.s_suppkey = __sq_2.ps_suppkey
+    LeftSemi Join: supplier.s_suppkey = __sq_1.ps_suppkey
```

Comment:
these changes imply the decorrelate passes used to be applied bottom up and after this PR they are applied top-down

Is that intentional? Or maybe I am misreading the diff 🤔

Reply:
😂 original implementation mix `TopDown` and `BottomUp`.
It use `TopDown` overall, but use `BottomUp` when match the subPlan.
I think it's a small mistake in original code but don't affect correctness.